knowledge-base/textarea-indicate-when-maxlength-is-reached.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Indicate that MaxLength is reached in TextArea
+description: How to indicate that MaxLength is reached in TextArea? How to display a message showing that the user has typed the maximum allowed characters? How to show a character counter?
+type: how-to
+page_title: Indicate that MaxLength is reached in TextArea
+slug: textarea-kb-indicate-when-maxlength-is-reached
+position: 
+tags: textarea, maxlength, max, count, counter, characters, message, indication
+ticketid:
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>TextArea for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+The TextArea has a `MaxLength` parameter but it does not show when the maximum allowed characters count is reached. How to notify the user they have reached the `MaxLength` value?
+
+How to display a counter, so the user knows how many characters they have left before reaching the `MaxLength` value?
+
+## Solution
+
+The `MaxLength` parameter of the TextArea maps to the `maxlength` attribute of the HTML `<textarea>` element. Its purpose is to prevent the user from typing if they reached the specified maximum allowed characters count. It does not provide visual indication out of the box.
+
+To notify the user that they have reached the `MaxLength` value:
+
+* Declare the desired message in an `if` block. 
+* Display this message if the length of the TextArea value equals the specified `MaxLength` value. 
+* Add the desired CSS rules or classes to style the message as needed.
+
+To let the user know how many characters they have left to type, render the TextArea value length and the `MaxLength` value - this will serve as a character counter.
+
+>caption Show a character counter and a message when max is reached
+
+````CSHTML
+@using System.ComponentModel.DataAnnotations
+
+@if (string.IsNullOrEmpty(FormSubmitMessage))
+{
+    <TelerikForm Model="@FormModel"
+                 OnValidSubmit="@OnValidSubmit"
```

Comment:
I can submit the form successfully even if I exceed the `MaxLength` value.

In general, this KB should only show how to implement a counter. There is no need to mimic validation errors with custom HTML, because our customers can achieve this with built-in DataAnnotations:

```
    public class ValidationModel
    {
        [Required(ErrorMessage = "Please enter a text.")]
        [MaxLength(40, ErrorMessage = "up to 40 please")]
        public string Text { get; set; }
    }
```